test/elm/convert/data.cql

```diff
@@ -122,3 +122,53 @@ define IsNull: ToBoolean('falsetto')
 // @Test: ToConcept
 define IsValid: ToConcept(Code { system: 'http://loinc.org', code: '8480-6' }) // Concept { codes: { Code { system: 'http://loinc.org', code: '8480-6' } } }
 define IsNull: ToConcept(null as Code)
+
+// @Test: ConvertsToBoolean
+define IsTrueWithTrueValue: ConvertsToBoolean('y')
+define IsTrueWithFalseValue: ConvertsToBoolean('0')
+define IsFalse: ConvertsToBoolean('foo')
+define IsNull: ConvertsToBoolean(null as String)
+
+// @Test: ConvertsToDate
+define IsTrue: ConvertsToDate('2015-01-01')
+define IsFalse: ConvertsToDate('foo')
+define IsNull: ConvertsToDate(null as Date)
+
+// @Test: ConvertsToDateTime
+define IsTrue: ConvertsToDateTime('2014-01-01T14:30:00.0Z')
+define IsFalse: ConvertsToDateTime('foo')
+define IsNull: ConvertsToDateTime(null as DateTime)
+
+// @Test: ConvertsToDecimal
+define IsTrue: ConvertsToDecimal('0.1')
+define IsFalse: ConvertsToDecimal('foo')
+define IsNull: ConvertsToDecimal(null as Decimal)
+
+// @Test: ConvertsToInteger
+define IsTrue: ConvertsToInteger('101')
+define IsFalse: ConvertsToInteger('foo')
+define IsNull: ConvertsToInteger(null as Integer)
+
+// @Test: ConvertsToQuantity
+define IsTrueWithDecimal: ConvertsToQuantity(0.1)
+define IsTrueWithInteger: ConvertsToQuantity(101)
+define IsTrueWithString: ConvertsToQuantity('1.0 \'mg\'')
+define IsFalse: ConvertsToQuantity('foo')
+define IsFalseWithInvalidUcum: ConvertsToQuantity('1.0 \'cc\'')
+define IsNull: ConvertsToQuantity(null as String)
+
+// @Test: ConvertsToRatio
+define IsTrue: ConvertsToRatio('1.0 \'mg\':2.0 \'mg\'')
+define IsFalse: ConvertsToRatio('1.0 \'mg\';2.0 \'mg\'')
+define IsFalseWithInvalidUcum: ConvertsToRatio('1.0 \'mg\';2.0 \'cc\'')
+define IsNull: ConvertsToRatio(null as String)
+
+// @Test: ConvertsToString
```

Comment:
There are additional signatures here that will likely fail for `Quantity`, `Ratio`, `Date`, `DateTime`, and `Time` unless we define a `toString()` function on them. It seems the `ToString` is similarly incomplete as well.

```
ConvertsToString(argument Boolean) Boolean
ConvertsToString(argument Integer) Boolean
ConvertsToString(argument Decimal) Boolean
ConvertsToString(argument Quantity) Boolean
ConvertsToString(argument Ratio) Boolean
ConvertsToString(argument Date) Boolean
ConvertsToString(argument DateTime) Boolean
ConvertsToString(argument Time) Boolean
```

Reply:
ToString relies on each type having a `toString()` function implemented. I have confirmed each of the listed types either have a built in `toString` or we have already implemented a custom function.